src/core/dfn-finder.js

```diff
@@ -145,20 +145,18 @@ function addAlternativeNames(dfn, names) {
 function findNormalDfn(defn, parent, ...names) {
   for (const name of names) {
     let resolvedName =
-      defn.type === "enum-value" && name === ""
-        ? "the-empty-string"
-        : name.toLowerCase();
-    let dfnForArray = definitionMap[resolvedName];
+      defn.type === "enum-value" && name === "" ? "the-empty-string" : name;
+    let dfnForArray = [...(definitionMap.get(resolvedName) || [])];
```

Comment:
Can we do:

```suggestion
    let dfnForArray = definitionMap.get(resolvedName) || [];
```

?

Reply:
It's a set as per dfn-map.js

Reply:
@sidvishnoi Hmm, in that case how about just do `definitionMap.get(resolvedName)` as we did before, instead of converting a set to an array every time? Line 156 then can be `dfnForArray && dfnForArray.size === 1` (or hopefully `dfnForArray?.size === 1` soon)

Reply:
(Yay no Marcos, everything is ours, party time 🎉)

Reply:
I think the motivation to use a Set comes from registerDefinition in dfn-map.js
https://github.com/w3c/respec/commit/8c0e56933356addf052c278a81860e9cec2ea280#diff-dbc281fb8c3069b68bed379e67cda5e4

Reply:
I think so too.

Reply:
Though we can move this dfnForArray inside the if block

Reply:
@saschanaz, hopefully less ugly now. Let me know what you think!